lcdblib/plotting/results_table.py

```diff
@@ -0,0 +1,1265 @@
+            # attach the dataframe itself and the index into that dataframe
+            # represented by the scatter points in the collection.
+            coll = self.marginal.scatter_ax.collections[-1]
+            coll.df = self.data
+            coll.ind = scatter_ind
+
+            color = color_converter(updated_kwargs['color'])
+            rug_x_kwargs['color'] = color
+            rug_y_kwargs['color'] = color
+
+            # Note: if both x and y are not valid, then they will not be on the
+            # plot at all (since there's no obvious place to put them).
+            items = [
+                # top rug, y is +inf and x is valid
+                (xi, ind & x_valid & y_is_pos_inf, pos_offset, rug_x_kwargs),
+
+                # one of the bottom rugs, where y is NaN
+                (xi, ind & x_valid & y_is_nan, nan_offset, rug_x_kwargs),
+
+                # bottom rug, y is -inf
+                (xi, ind & x_valid & y_is_neg_inf, neg_offset, rug_x_kwargs),
+
+                # right rug, x is +inf
+                (yi, ind & y_valid & x_is_pos_inf, pos_offset, rug_y_kwargs),
+
+                # one of the left rugs; x is NaN
+                (yi, ind & y_valid & x_is_nan, nan_offset, rug_y_kwargs),
+
+                # left rug, x is -inf
+                (yi, ind & y_valid & x_is_neg_inf, neg_offset, rug_y_kwargs),
+            ]
+            for values, index, offset, kwargs in items:
+                coll = EventCollection(
+                    values[index], lineoffset=offset, **kwargs)
+                coll.df = self.data
+                coll.ind = index
+                ax.add_collection(coll)
+
+            # Plot the text names if configured
+            if names:
+                transOffset = matplotlib.transforms.offset_copy(
+                    ax.transData, fig=ax.figure, **offset_kwargs)
+
+                for xii, yii, name in zip(xi[ind], yi[ind], names):
+                    ax.text(xii,
+                            yii,
+                            name,
+                            transform=transOffset,
+                            **label_kwargs)
+
+        # register callback
+        if callback is None:
+            callback = self._default_callback
+
+        # This trick is so that we can say, in the docstring, that a callback
+        # function must accept an index name (i.e. gene ID). Much easier than
+        # saying "the callback function accepts an event, and the event's
+        # artist has an `ind` and `df` attribute".
+        def wrapped_callback(event):
+            for _id in self._id_callback(event):
+                callback(_id)
+
+        # Connect the callback.
+        ax.figure.canvas.mpl_connect('pick_event', wrapped_callback)
+
+        ax.set_xlabel(xlab)
+        ax.set_ylabel(ylab)
+        ax.axis('tight')
+
+        return ax
+
+    def radviz(self, column_names, transforms=dict(), **kwargs):
+        """
```

Comment:
The pandas function does a lot of stuff on its own, and I wanted to be able to do things like attach callbacks (for clicking on points in the callback) and to provide a similar API as to `ResultsTable.scatter`.